### PR TITLE
Run kfctl go unittests as part of pre/post submittests

### PR DESCRIPTION
**Pull request by jlewi**
* Run the kfctl unittests as part of pre/postsubmit s

* Add a stage to Dockerfile to create an environment suitable for building and
testing kfctl; don't include any source in this stage.

* Create a .gcloudignore file to exclude directories that shouldn't be uploaded
  to GCB when building with GCB.

* Create a cloudbuild.yaml file to build the docker images.
  * We can use the docker --target flag to specify which stage we want to build

* Update the developer guide; add instructions on how to run the tests
  locally in a docker image.

Related to #2874 run kfctl unittests as part of ci.

Fix the hasDefaultStorage class unittest.

  * It doesn't look like the test would ever have passed; the test
    was setting the parameters field and not the annotations on the storage
    class.

* Comment out some unittests that aren't passing; file #3056

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3052)

### CODE REVIEW

**Comment by jlewi** [comment]
/assign @gabrielwen 
/assign @kkasravi 

This is should be ready the most recent test failure looks like kfctl just died during delete and should be unrelated to this PR.

**Comment by gabrielwen** [comment]
/lgtm

I'm not export to those build configs.  will let @kkasravi make further review on it.  Test results looks normal.

File: bootstrap/developer_guide.md [hunk 1]
```diff
@@ -87,6 +96,38 @@ make build-kfctl
...
+
+Alternatively to build using GCB
+
+```
+ GCLOUD_PROJECT=cloud-ml-dev make build-builder-container
+```
+
+We can now start an interactive shell inside the docker container
+
+```
```
Comment (kkasravi) [inline bootstrap/developer_guide.md]: add
```
IMAGE=$(docker images --format '{{ .Repository }}:{{ .Tag }}' | awk 'FNR==1{print $0}')
```
above REPO_ROOT

File: bootstrap/developer_guide.md [hunk 2]
```diff
@@ -87,6 +96,38 @@ make build-kfctl
...
+REPO_ROOT=$(git rev-parse --show-toplevel)
+mkdir -p ${REPO_ROOT}/bootstrap/bin/tmp
+docker run -it \
```
Comment (kkasravi) [inline bootstrap/developer_guide.md]: replace with 
```
docker run  -v ${REPO_ROOT}:/go/src/github.com/kubeflow/kubeflow  -v ${REPO_ROOT}/bootstrap/bin/tmp:/tmp -it $IMAGE /bin/bash -c "cd /go/src/github.com/kubeflow/kubeflow/bootstrap;exec /bin/bash"
```

Comment (kkasravi) [reply bootstrap/developer_guide.md]: I didn't see a `make test` anywhere...

Comment (jlewi) [reply bootstrap/developer_guide.md]: Done

File: bootstrap/developer_guide.md (hunk 1 above)
Comment (jlewi) [reply bootstrap/developer_guide.md]: Done

**Pull request merged**